src/CommandAnalysis.cc

```diff
@@ -463,13 +467,63 @@ void CommandAnalysis::evaluate(const MemorySpecification& memSpec,
       if (mem_state != CommandAnalysis::MS_SREF) {
         cerr << "Incorrect use of Self-Refresh Power-Up!" << endl;
       }
-      if (max(zero, timestamp - sref_cycle) >= memSpec.memTimingSpec.RFC) {
-        sref_cycles         += max(zero, timestamp - sref_cycle
-                                   - memSpec.memTimingSpec.RFC);
+
+      // The total duration of self-refresh is given by the difference between
+      // the current clock cycle and the clock cycle of entering self-refresh.
+      int64_t sref_duration = timestamp - sref_cycle;
+
+      // Negative or zero duration should never happen.
+      assert(sref_duration > 0);
```

Comment:
We have never punished users that generate a broken command trace with a hard assert. Perhaps it is better to print a warning here? We could consider adding a flag in a future version that turns warning prints into asserts.